src/rez/developer_package.py

```diff
@@ -152,6 +170,14 @@ def get_reevaluated(self, objects):
         with set_objects(objects):
             return self.from_path(self.root)
 
+    # TODO: should have a better name
+    def evaluate_directives(self, build_context, variant_index=None):
+        variant = self.get_variant(variant_index)
+        override = evaluate_directive_requires(variant, build_context)
+        # TODO: if no directive, skip re-evaluate variant
+        package = self._from_path(self.root, override=override)
+        return package.get_variant(variant_index)
+
```

Comment:
Not sure if there is a better way to re-apply resolved directives back to variant.
I have tried to override the variant data/resource but failed, it looks like we must re-evaluate it again with passing resolved directive request as *override*.